pkg/kubectl/deployment.go

```diff
@@ -100,80 +154,30 @@ func (s *DeploymentBasicGeneratorV1) StructuredGenerate() (runtime.Object, error
 				Spec: podSpec,
 			},
 		},
-	}
-	return &deployment, nil
-}
-
-// validate validates required fields are set to support structured generation
-func (s *DeploymentBasicGeneratorV1) validate() error {
-	if len(s.Name) == 0 {
-		return fmt.Errorf("name must be specified")
-	}
-	if len(s.Images) == 0 {
-		return fmt.Errorf("at least one image must be specified")
-	}
-	return nil
+	}, err
 }
 
 // DeploymentBasicAppsGeneratorV1 supports stable generation of a deployment under apps/v1beta1 endpoint
 type DeploymentBasicAppsGeneratorV1 struct {
-	Name   string
-	Images []string
+	BaseDeploymentGenerator
 }
 
 // Ensure it supports the generator pattern that uses parameters specified during construction
 var _ StructuredGenerator = &DeploymentBasicAppsGeneratorV1{}
 
-func (DeploymentBasicAppsGeneratorV1) ParamNames() []GeneratorParam {
-	return []GeneratorParam{
-		{"name", true},
-		{"image", true},
-	}
-}
-
 func (s DeploymentBasicAppsGeneratorV1) Generate(params map[string]interface{}) (runtime.Object, error) {
-	err := ValidateParams(s.ParamNames(), params)
+	base, err := baseDeploymentGeneratorFromParams(params)
 	if err != nil {
 		return nil, err
 	}
-	name, isString := params["name"].(string)
-	if !isString {
-		return nil, fmt.Errorf("expected string, saw %v for 'name'", name)
-	}
-	imageStrings, isArray := params["image"].([]string)
-	if !isArray {
-		return nil, fmt.Errorf("expected []string, found :%v", imageStrings)
-	}
-	delegate := &DeploymentBasicAppsGeneratorV1{Name: name, Images: imageStrings}
-	return delegate.StructuredGenerate()
+	return (&DeploymentBasicAppsGeneratorV1{*base}).StructuredGenerate()
 }
 
 // StructuredGenerate outputs a deployment object using the configured fields
 func (s *DeploymentBasicAppsGeneratorV1) StructuredGenerate() (runtime.Object, error) {
-	if err := s.validate(); err != nil {
-		return nil, err
-	}
-
-	podSpec := v1.PodSpec{Containers: []v1.Container{}}
-	for _, imageString := range s.Images {
-		// Retain just the image name
-		imageSplit := strings.Split(imageString, "/")
-		name := imageSplit[len(imageSplit)-1]
-		// Remove any tag or hash
-		if strings.Contains(name, ":") {
-			name = strings.Split(name, ":")[0]
```

Comment:
Looking at this closer, it seems weird that we were stripping the tags.  Aren't these important information?